### ISSUE TITLE
Goal of this repo

### ISSUE THREAD

**Issue by giltho** [semgrep/ocaml-tree-sitter-core#82]
Hello, I have a few questions about this repo
- This repo does not provide bindings to tree-sitter right? Only a way of generating OCaml parsers from a tree-sitter syntax?
- Is there a way to extract such a parser to use?
- Is there any LICENSE? Am I allowed to just copy code, since it's not published on OPAM?
- If I want an incremental parser for C in OCaml based on the tree-sitter syntax, but I do not care about the semgrep-related feature, should I use semgrep-c anyway?
- Is there any kind of documentation on how to drive the generated parsers?

Basically, if I want to use tree-sitter in OCaml to incrementally parse programs (to write, e.g., LSP servers), should I be using this repo, or creating my own bindings?
Thanks!

**Comment by giltho** [comment]
I read some more code, I have some answers to my own questions:

> This repo does not provide bindings to tree-sitter right? Only a way of generating OCaml parsers from a tree-sitter syntax?

After reading again, I think the additional parsing stuff is just semgrep-specific, but the language parser is done through the tree-sitter bindings, so all good.

I tried installing things locally again, but things don't compile yet, I'll do more attempts before giving more details.

I think the most important questions left is: do you consider the generated code to be licensed?

**Comment by aryx** [comment]
No, the generated code is not licensed.
